libraries/AP_Scripting/AP_Scripting_CANSensor.cpp

```diff
@@ -62,8 +63,24 @@ bool ScriptingCANBuffer::read_frame(AP_HAL::CANFrame &frame)
 // recursively add frame to buffer
 void ScriptingCANBuffer::handle_frame(AP_HAL::CANFrame &frame)
 {
+    // Check if frame matches any filters
+    bool accept = num_filters == 0;
+    static_assert(sizeof(filter_mask) == sizeof(filter_value), "filter mask and values must match");
+    for (uint8_t i = 0; i < MIN(num_filters, sizeof(filter_mask)); i++) {
```

Comment:
```suggestion
    for (uint8_t i = 0; i < MIN(num_filters, ARRAY_SIZE(filter_mask)); i++) {
```
My bad, we want the array length not the object size.